src/backend/joanie/tests/core/test_api_admin_course_runs.py

```diff
@@ -233,3 +234,129 @@ def test_admin_api_course_runs_delete(self):
         response = self.client.delete(f"/api/v1.0/admin/course-runs/{course_run.id}/")
 
         self.assertEqual(response.status_code, 204)
+
+    def test_admin_api_course_run_read_list_authenticated_with_nested_course_filter_name(
+        self,
+    ):
+        """
+        When filtering by name the query returns only relevant CourseRuns
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        course = factories.CourseFactory.create()
+        course_run = factories.CourseRunFactory(
+            course=course, is_listed=True, title="This is a test"
+        )
+        factories.CourseRunFactory(course=course, is_listed=True, title="CourseRun 0")
+        factories.CourseRunFactory(course=course, is_listed=True, title="CourseRun 1")
+        factories.CourseRunFactory(course=course, is_listed=True, title="CourseRun 2")
+
+        response = self.client.get(
+            f"/api/v1.0/admin/courses/{course.id}/course-runs/?query=is%20a",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["count"], 1)
+        self.assertEqual(content["results"][0]["id"], str(course_run.id))
+        response = self.client.get(
+            f"/api/v1.0/admin/courses/{course.id}/course-runs/?query=Cour",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        content = response.json()
+        self.assertEqual(content["count"], 3)
+
+    def test_admin_api_course_run_read_list_authenticated_with_nested_course_filter_start_date(
+        self,
+    ):
+        """
+        When filtering by date the query returns only CourseRuns starting at
+        or after the given date
+        """
+        admin = factories.UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=admin.username, password="password")
+        now = datetime.utcnow()
+        course = factories.CourseFactory.create()
+        course_run = factories.CourseRunFactory(
+            course=course, is_listed=True, start=now
+        )
+        factories.CourseRunFactory(
+            course=course, is_listed=True, start=now - timedelta(weeks=20)
+        )
+
```

Comment:
I think you should create one course run in each of the possible states to test each of your filterings at least once?

If you wait for [this PR](https://github.com/openfun/joanie/pull/369) it will be nice and much easier to do it?

Reply:
This test is the one that checks filtering by date, the filter by state starts at line 312 and I already created a CourseRun for each state, plus a batch of 20 random ones to be sure